src/main/java/com/snowflake/snowpark_java/types/Variant.java

```diff
@@ -376,6 +376,27 @@ public String asJsonString() {
     }
   }
 
+  /**
+   * Converts the variant as valid JsonNode. This function allows to read the JSON object directly
+   * rather parsing it as String and then extract the desired value
+   *
+   * <pre>{@code - to get the first value from array for key "a"
+   *
+   * Variant jv = new Variant("{\"a\": [1, 2], \"b\": \"c\"}");
+   * JsonNode jNode = jv.asJsonNode();
+   * System.out.println(jNode.get("a").get(0));
+   *
+   * output
+   * 1
+   * }</pre>
+   *
+   * @return A valid JsonNode
+   * @since 1.14.0
+   */
+  public JsonNode asJsonNode() {
+    return objectToJsonNode(value);
```

Comment:
I think you are missing the case where the value is just a JsonNode.

Reply:
The idea is to avoid any overhead. So if value is already a JsonNode we should be able to just return it.